### PR TITLE
JDK-8284858: Start of release updates for JDK 20

### PR DESCRIPTION
**Pull request by jddarcy**
Time to start getting ready for JDK 20...

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.java.net/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issues
 * [JDK-8284858](https://bugs.openjdk.org/browse/JDK-8284858): Start of release updates for JDK 20
 * [JDK-8286035](https://bugs.openjdk.org/browse/JDK-8286035): Add source 20 and target 20 to javac
 * [JDK-8286034](https://bugs.openjdk.org/browse/JDK-8286034): Add SourceVersion.RELEASE_20

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [54e872b5](https://git.openjdk.java.net/jdk/pull/8236/files/54e872b5f3660f0909a74a0f43b144dc04fc02fd)
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - Author) ⚠️ Review applies to [96be1787](https://git.openjdk.java.net/jdk/pull/8236/files/96be178730ff30be3d98384456a9889a97a80764)
 * [Iris Clark](https://openjdk.java.net/census#iris) (@irisclark - **Reviewer**)
 * [Erik Joelsson](https://openjdk.java.net/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to [96be1787](https://git.openjdk.java.net/jdk/pull/8236/files/96be178730ff30be3d98384456a9889a97a80764)
 * [Jonathan Gibbons](https://openjdk.java.net/census#jjg) (@jonathan-gibbons - **Reviewer**) ⚠️ Review applies to [eedd36bd](https://git.openjdk.java.net/jdk/pull/8236/files/eedd36bd5edc7125e085012e5bb9a33129335c6c)
 * [Magnus Ihse Bursie](https://openjdk.java.net/census#ihse) (@magicus - **Reviewer**) ⚠️ Review applies to [e495a579](https://git.openjdk.java.net/jdk/pull/8236/files/e495a579eb288970626c7d782d4353c568845a69)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8236/head:pull/8236` \
`$ git checkout pull/8236`

Update a local copy of the PR: \
`$ git checkout pull/8236` \
`$ git pull https://git.openjdk.java.net/jdk pull/8236/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8236`

View PR using the GUI difftool: \
`$ git pr show -t 8236`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8236.diff">https://git.openjdk.java.net/jdk/pull/8236.diff</a>

</details>

### CODE REVIEW

**Comment by jddarcy** [comment]
The expected kinds of updates to start up JDK 20.

File: src/java.base/share/classes/jdk/internal/org/objectweb/asm/Opcodes.java [hunk 1]
```diff
@@ -309,6 +309,7 @@ public interface Opcodes {
     int V17 = 0 << 16 | 61;
     int V18 = 0 << 16 | 62;
     int V19 = 0 << 16 | 63;
+    int V20 = 0 << 17 | 64;
```
Comment (dholmes-ora) [inline src/java.base/share/classes/jdk/internal/org/objectweb/asm/Opcodes.java:312]: 17 ??

**Comment by kevinrushforth** [comment]
You also need to change the JBS version from 19 to 20 in [`.jcheck/conf`](https://github.com/openjdk/jdk/blob/6a33974a6b8a629744c6d76c3b4fa1f772e52ac8/.jcheck/conf#L4):

**Review by dholmes-ora — APPROVED** [review]
One comment below.
I ignored the sym files.
Everything else appears okay.
Thanks.

File: src/java.base/share/classes/jdk/internal/org/objectweb/asm/Opcodes.java (hunk 1 above)
Comment (jddarcy) [reply src/java.base/share/classes/jdk/internal/org/objectweb/asm/Opcodes.java:312]: Doh! Will fix in the next pushed.

**Comment by jddarcy** [comment]
> You also need to change the JBS version from 19 to 20 in [`.jcheck/conf`](https://github.com/openjdk/jdk/blob/6a33974a6b8a629744c6d76c3b4fa1f772e52ac8/.jcheck/conf#L4):

Acknowledged; will fix in the next push. Thanks.

**Review by kevinrushforth — APPROVED** [review]

**Review by irisclark — APPROVED** [review]

**Review by dholmes-ora — APPROVED** [review]

**Review by erikj79 — APPROVED** [review]

**Review by irisclark — APPROVED** [review]

**Review by irisclark — APPROVED** [review]

**Review by jonathan-gibbons — APPROVED** [review]
langtools files look OK, although I did skim through the auto-generated new data/symbols files.

There are possibilities in the langtools file to simplify naming and default initialization of member fields in various places, in a separate PR, separate from these changes.

**Review by dholmes-ora — APPROVED** [review]
Updates look good. Thanks

**Review by irisclark — APPROVED** [review]

**Pull request closed without merging**